src/utils/scripts.ts

```diff
@@ -37,3 +39,87 @@ export const parseP2PKH = (buff: Buffer, network: Network): P2PKH => {
 
   return new P2PKH(helpers.encodeAddress(addressHash, network), { timelock });
 };
+
+/**
+* Parse Data output script
+*
+* @param {Buffer} buff Output script
+*
+* @return {ScriptData} ScriptData object
+*/
+export const parseScriptData = (buff: Buffer): ScriptData => {
+  // We should clone the buffer being sent in order to never mutate
+  // what comes from outside the library
+  let scriptBuf = _.clone(buff);
+  if (scriptBuf.length < 2) {
+    // At least 1 byte for len data and 1 byte for OP_CHECKSIG
+    throw new ParseScriptError('Invalid output script.');
+  }
+
+  // The expected len will be at least 2 bytes
+  // 1 for the script len and 1 for the OP_CHECKSIG in the end
+  let expectedLen = 2;
+  let dataBytesLen;
+
+  // If we have OP_PUSHDATA1 as first byte, the second byte has the length of data
+  // otherwise, the first byte already has the length of data
+  if (scriptBuf[0] === OP_PUSHDATA1[0]) {
+    expectedLen += 1;
+    dataBytesLen = scriptBuf[1];
```

Comment:
Should we have a test for this case?

Reply:
Agreed. Added here https://github.com/HathorNetwork/hathor-wallet-lib/pull/295/commits/b7bf56105ac7290106987153ba64e1e302821ee5